sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -153,14 +153,24 @@ private object CosmosWriteConfig {
     parseFromStringFunction = bulkEnabledAsString => bulkEnabledAsString.toBoolean,
     helpMessage = "Cosmos DB Item Write bulk enabled")
 
-  private val writeMaxConcurrency = 100
+  // TODO: moderakh this should be dynamic based on the number of CPU cores
+  private val MaxPointWriteConcurrency = 100
```

Comment:
NIT: Why not just do it in this PR? (I mean some starting point with multiple of Runtme.getRuntime().availableProcessors - will probably need adjustment later anyway)

Reply:
sure done.